xcov19/tests/tests_mixin.py

```diff
@@ -0,0 +1,40 @@
+import pytest
+from xcov19.utils.mixins import InterfaceProtocolCheckMixin
+
+
+class BaseClass:
+    @classmethod
+    def method_with_params(cls, param1: int, param2: str) -> None:
+        pass
+
+
```

Comment:
You could simply move the setup before actual tests so the definitions would be clearer to read and your code would only instantiate it.

```suggestion

class IncorrectImplementation(BaseClass, InterfaceProtocolCheckMixin):
            """Does not implement method_with_params"""
            pass
```